contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,362 @@
+        }
+
+        /// Returns human-readable string of the Cadence code.
+        ///
+        access(all) view fun codeAsCadence(): String {
+            return String.fromUTF8(self.code.decodeHex()) ?? panic("Problem stringifying code!")
+        }
+
+        /// Replaces the ContractUpdate code with that provided.
+        ///
+        access(contract) fun replaceCode(_ code: String) {
+            self.code = code
+        }
+    }
+
+    /********************
+            Host
+     ********************/
+
+    /// Serves as identification for a caller's address.
+    /// NOTE: Should be saved in storage and access safeguarded as reference grants access to contract staging. If a
+    /// contract host wishes to delegate staging to another account (e.g. multisig account setup enabling a developer
+    /// to stage on its behalf), it should create a PRIVATE Host capability and publish it to the receiving account.
+    ///
+    access(all) resource Host {
+        /// Returns the resource owner's address
+        ///
+        access(all) view fun address(): Address {
+            return self.owner?.address ?? panic("Host is unowned!")
+        }
+    }
+
+    /********************
+            Capsule
+     ********************/
+
+    /// Resource that stores pending contract updates in a ContractUpdate struct. On staging a contract update for the
+    /// first time, a Capsule will be created and stored in this contract account. Any time a stageContract() call is
+    /// made again for the same contract, the code in the Capsule will be replaced. As you see, the Capsule is merely
+    /// intended to store the code, as contract updates will be executed by state migration across the network at the
+    /// Cadence 1.0 milestone.
+    ///
+    access(all) resource Capsule {
+        /// The address, name and code of the contract that will be updated.
+        access(self) let update: ContractUpdate
+
+        init(update: ContractUpdate) {
+            pre {
+                update.codeAsCadence() != nil: "Staged update code must be valid Cadence"
+                update.isValid(): "Target contract does not exist"
+            }
+            self.update = update
+        }
+
+        /// Returns the staged contract update in the form of a ContractUpdate struct.
+        ///
+        access(all) view fun getContractUpdate(): ContractUpdate {
+            return self.update
+        }
+
+        /// Replaces the staged contract code with the given hex-encoded Cadence code.
+        ///
+        access(contract) fun replaceCode(code: String) {
+            post {
+                self.update.codeAsCadence() != nil: "Staged update code must be valid Cadence"
+            }
+            self.update.replaceCode(code)
+            emit StagingStatusUpdated(
+                capsuleUUID: self.uuid,
+                address: self.update.address,
+                codeHash: code.decodeHex(),
```

Comment:
See above:
```suggestion
                codeHash: code,
```

Reply:
Are you suggesting here that the event codeHash type be updated from String to [UInt8]?